src/daft-functions/src/temporal/mod.rs

```diff
@@ -135,6 +135,60 @@ pub fn dt_time(input: ExprRef) -> ExprRef {
     ScalarFunction::new(Time {}, vec![input]).into()
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct UnixTimestamp {
+    pub time_unit: TimeUnit,
+}
+
+#[typetag::serde]
+impl ScalarUDF for UnixTimestamp {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn name(&self) -> &'static str {
+        "to_unix_epoch"
+    }
+
+    fn to_field(&self, inputs: &[ExprRef], schema: &Schema) -> DaftResult<Field> {
+        match inputs {
+            [input] => match input.to_field(schema) {
+                Ok(field) => match field.dtype {
+                    DataType::Timestamp(..) | DataType::Date => {
+                        Ok(Field::new(field.name, DataType::Int64))
+                    }
+                    _ => Err(DaftError::TypeError(format!(
+                        "Expected input to be date or timestamp, got {}",
+                        field.dtype
+                    ))),
+                },
+                Err(e) => Err(e),
+            },
+            _ => Err(DaftError::SchemaMismatch(format!(
+                "Expected 1 input arg, got {}",
+                inputs.len()
+            ))),
+        }
+    }
+
+    fn evaluate(&self, inputs: &[Series]) -> DaftResult<Series> {
+        match inputs {
+            [input] => input
+                .cast(&DataType::Timestamp(self.time_unit, None))
+                .and_then(|s| s.cast(&DataType::Int64)),
```

Comment:
Why the double cast vs downcast the series? Are we leveraging a up/down cast time unit precisions from arrow2 and do they behave as we want?

Reply:
so the idea was that if we cast to a timestamp with the given unit, the inner representation is the `unix_epoch`. This is because the timestamp logical type is backed by `i64` physical type. So we can cheaply cast `ts` -> `i64` to extract the underlying value

We could write a specialized kernel for this, but this does the same thing, and the casts are already pretty optimized.

Reply:
we don't want to just downcast because it could be in a different timeunit, or it could also be a `date` datatype. so we actually want a materialized cast to the timestamp to ensure the underlying `i64` is correct.